ts/features/itwallet/common/components/ItwCredentialCard.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Badge,
+  Body,
+  HSpacer,
+  IOColors,
+  Tag
+} from "@pagopa/io-app-design-system";
+import React from "react";
+import { Image, ImageSourcePropType, StyleSheet, View } from "react-native";
+import I18n from "../../../../i18n";
+import { CredentialType } from "../utils/itwMocksUtils";
+
+export type ItwCredentialStatus = "valid" | "pending" | "expiring" | "expired";
+
+type PreviewProps = {
+  isPreview: true;
+  data?: never;
+};
+
+type DataProps = {
+  isPreview?: false;
+  data: ReadonlyArray<string>;
+};
+
+type BaseProps = {
+  credentialType: CredentialType;
+  isMasked?: boolean;
+  status?: ItwCredentialStatus;
+};
+
+export type ItwCredentialCard = BaseProps & (PreviewProps | DataProps);
+
+export const ItwCredentialCard = ({
+  isMasked = false,
+  status = "valid",
+  credentialType,
+  ...props
+}: ItwCredentialCard) => {
+  const isValid = status === "valid";
+  const shouldDisplayData = !(!isValid || isMasked) && !props.isPreview;
+  const labelColor: IOColors = isValid ? "bluegreyDark" : "grey-700";
+
+  const cardBackgroundSource =
+    credentialCardBackgrounds[credentialType][isValid ? 0 : 1];
+  const statusTagProps = tagPropsByStatus[status];
+
+  return (
+    <View style={props.isPreview && styles.previewContainer}>
+      <View style={styles.cardContainer}>
+        <View style={styles.card}>
+          <Image
+            style={{ height: "100%", width: "100%" }}
```

Comment:
What do you think to use a styles prop?

Reply:
Addressed in https://github.com/pagopa/io-app/pull/5832#:~:text=styles%20to%20stylesheets-,8b33b8a,-mastro993%20requested%20a